Kernel/KBO.cpp

```diff
@@ -93,47 +102,36 @@ class KBO::State
  * traverse(Term*,int) for both terms/literals in case their
  * top functors are different.)
  */
-Ordering::Result KBO::State::result(Term* t1, Term* t2)
+Ordering::Result KBO::State::result(const AppliedTerm& t1, const AppliedTerm& t2)
 {
```

Comment:
This seems like a case where it might be worth having `Applied<Term *>` versus `Applied<TermList>`.

Reply:
Your call - just an observation.